core/src/main/java/org/apache/druid/java/util/common/guava/ParallelMergeCombiningSequence.java

```diff
@@ -575,21 +617,22 @@ protected void compute()
               adjustedNextYieldAfter,
               batchSize,
               targetTimeNanos,
-              recursionDepth + 1,
+              metricsAccumulator,
               cancellationGizmo
           ));
         } else if (cancellationGizmo.isCancelled()) {
           // if we got the cancellation signal, go ahead and write terminal value into output queue to help gracefully
           // allow downstream stuff to stop
-          LOG.debug("cancelled after %s tasks", recursionDepth);
+          LOG.debug("cancelled after %s tasks", metricsAccumulator.getTaskCount());
           outputQueue.offer(ResultBatch.TERMINAL);
         } else {
           // if priority queue is empty, push the final accumulated value into the output batch and push it out
           outputBatch.add(currentCombinedValue);
+          metricsAccumulator.incrementOutputRows(batchCounter + 1L);
```

Comment:
Is the 1L here for the terminal value? Does that need to be counted towards the output rows, since it's not really a row?

Reply:
No, it is for the straggling `currentCombinedValue` that is being added to the batch the line before, which comes from here if the inputs were exhausted: https://github.com/apache/incubator-druid/blob/master/core/src/main/java/org/apache/druid/java/util/common/guava/ParallelMergeCombiningSequence.java#L537